lib/RexStanSettings.php

```diff
@@ -0,0 +1,250 @@
+                $value = RexStanUserConfig::getPaths();
+                break;
+            case 'phpVersion':
+                $value = RexStanUserConfig::getPhpVersion();
+                break;
+            case 'includes':
+                $value = RexStanUserConfig::getIncludes();
+                break;
+
+                // Default: als rex_config-Wert behandeln
+            default:
+                $value = parent::getValue($name);
+        }
+
+        return $value;
+    }
+
+    /**
+     * Speichert die Änderungen des Settings-Formuars.
+     *
+     * Die Parent-Methode schreibt die Werte in die Tabelle rex_config.
+     * Hier werden zunächst die für phpstan vorgesehenen Werte abgefangen und in die $userConf
+     * übertragen. Am Ende wird $userConf auf die Platte geschrieben.
+     * Alle übrigen Felder werden als rex_config gespeichert.
+     */
+    protected function save(): bool
+    {
+    
+        // für alle Fälle
+        $level = RexStanUserConfig::getLevel();
+        $paths = RexStanUserConfig::getPaths();
+        $phpVersion = RexStanUserConfig::getPhpVersion();
+        $includes = RexStanUserConfig::getIncludes();
+    
+        foreach ($this->getSaveElements() as $fieldsetElements) {
+            foreach ($fieldsetElements as $element) {
+                // read-only-fields nicht speichern
+                if ($element->isReadOnly()) {
+                    continue;
+                }
+
+                $fieldName = $element->getFieldName();
+                $fieldValue = $element->getSaveValue();
+
+                if (is_string($fieldValue)) {
+                    $fieldValue = trim($fieldValue);
+                }
+
+                switch ($fieldName) {
+                    // Zunächst die phpstan-Parameter auslesen und ggf aufbereiten
+                    case 'level':
+                        $level = (int) $fieldValue;
+                        break;
+                    case 'paths':
+                        $paths = array_filter(explode('|', trim((string) $fieldValue, '|')), 'strlen'); // String |x|y| rückumwandeln in [x,y]
+                        break;
+                    case 'phpVersion':
+                        $phpVersion = (int) $fieldValue;
+                        break;
+                    case 'includes':
+                        $includes = array_filter(explode('|', trim((string) $fieldValue, '|')), 'strlen'); // String |x|y| rückumwandeln in [x,y]
+                        break;
+
+                        // Default: als rex_config-Wert behandeln
+                    default:
+                        rex_config::set('rexstan', $fieldName, $fieldValue);
+                }
+            }
+        }
+
+        // Für die phpstan-Konfigurationseinträge: separat sichern
+        RexStanUserConfig::save($level, $paths, $includes, $phpVersion);
+        
+        // Just a notification via EP
+        rex_extension::registerPoint(new rex_extension_point('REXSTAN_SETTINGS_SAVED', '', []));
```

Comment:
bisher ist mir kein use-case für nen EP bekannt, daher würde ich keinen anlegen

Reply:
Es wäre das Äquivalent zu REX_FORM_SAVED aus der Klasse rex_form, denn rex_config_form hat keinen solchen EP. Fürmich wäre der use case, dass man ggf. auch den Code-Analyser im Edititor (bei mir PHPIntellephense in VSCode) mit umstellen könnte. Ja, ist etwas akademisch; ich weiß aktuell auch nicht, ob ich es tatsächlich nutzen würde. An dem Thema häng ich nicht, andere Sachen sind mir wichtiger.